cuqi/experimental/geometry/_productgeometry.py

```diff
@@ -0,0 +1,163 @@
+        return len(self.geometries)
+    
+    def _split_par(self, par):
+        """Splits a stacked parameter vector into parameter vectors for each
+        geometry."""
+        return tuple(np.split(par, self.stacked_par_split_indices))
+    
+    def _plot(self, values, **kwargs):
+        """Plotting function for the product geometry."""
+        raise NotImplementedError(
+            f"Plotting not implemented for {self.__class__.__name__}.")
+
+    def par2fun(self, *args):
+        """Converts parameter vector(s) into function values for each 
+        geometry. The parameter vector can be stacked (all parameters are
+        in one vector) or unstacked (one parameter vector corresponds to
+        each geometry)."""
+
+        # If one argument is passed, then it is assumed that the parameter
+        # vector is stacked and split it.
+        # No effect if the parameter vector is already split and corresponds
+        # to one geometry.
+        if len(args) == 1:
+            args = self._split_par(args[0])
+
+        # Convert parameter vectors to function values for each geometry.
+        fun_list = []
+        for i, g in enumerate(self.geometries):
+            fun_i = g.par2fun(args[i])
+            fun_list.append(fun_i)
+        return tuple(fun_list)
+
+    def fun2par(self, *funvals, stacked=False):
+        """Converts (multiple) function values into the corresponding
+        parameter vectors. If the flag stacked is set to True, then the
+        parameter vectors are stacked into one vector. Otherwise, the
+        parameter vectors are returned as a tuple."""
+
+        par_vec = []
+        for i, g in enumerate(self.geometries):
+            par_i = g.fun2par(funvals[i])
+            par_vec.append(par_i)
+
+        # stack parameters:
+        if stacked:
+            # if single sample
+            if len(par_vec[0].shape) == 1:
+                stacked_val = np.hstack(par_vec)
+            elif len(par_vec[0].shape) == 2:
+                stacked_val = np.vstack(par_vec)
+            else:
+                raise ValueError(
+                    "Cannot stack parameter vectors with more than 2 dimensions."
+                    )
+
+        return  stacked_val if stacked else tuple(par_vec)
+
+    def vec2fun(self, *funvec):
+        """Maps function vector representation, if available, to function 
+        values."""
+        fun_values = []  
+        for i, g in enumerate(self.geometries):
+            fun_values.append(g.vec2fun(funvec[i]))
+
+        return tuple(fun_values)
+    
+    def fun2vec(self, fun):
+        """Maps function values to a vector representation of the function values, if available."""
+        vec_values = []
+        for i, g in enumerate(self.geometries):
+            vec_values.append(g.fun2vec(fun[i]))
+        
+        return tuple(vec_values)
+        
+    
+    def __repr__(self) -> str:
+        """Representation of the product geometry."""
```

Comment:
Printing looks a little off?
![image](https://github.com/user-attachments/assets/d11af2b4-8dd5-4744-84e4-7ffbfa155793)
not sure we need this "of the following geometries"